packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -44,13 +55,14 @@ export const UPlotChart: React.FC<PlotProps> = props => {
   }, [plotInstance]);
 
   useLayoutEffect(() => {
+    const data = props.data.frame.fields.map(f => f.values.toArray()) as AlignedData;
+    const shouldUpdate = !isEqual(plotData?.current?.data, data);
```

Comment:
This is a pottential bottleneck

Reply:
not sure I understand this, if we are in this useLayoutEffect data has always changed, should be no need to do this and this could be very expensive no? 

of should be super easy to just check if data is different from previous? Ah I see this function changed from last I edited, it used to only depend on data, now on data and config. Then we need to track previous data as well so we can just do a instance difference check, should be no need for isEqual here, just normal !== check from previous